pkg/healthcheck/microshift_core_workloads.go

```diff
@@ -68,11 +84,58 @@ func getExpectedCSIComponents(cfg *config.Config) []string {
 	deployments := []string{}
 	for _, comp := range cfg.Storage.OptionalCSIComponents {
 		if comp == config.CsiComponentSnapshot {
+			klog.Infof("CSI Snapshot Controller is enabled")
 			deployments = append(deployments, "csi-snapshot-controller")
 		}
 		if comp == config.CsiComponentSnapshotWebhook {
+			klog.Infof("CSI Snapshot Webhook is enabled")
 			deployments = append(deployments, "csi-snapshot-webhook")
 		}
 	}
 	return deployments
 }
+
+// optionalComponents checks for existence of namespaces that are deployed
+// using MicroShift's optional RPMs. If the namespace exists, the workloads
+// are added to the map of expected readiness.
+func optionalComponents(ctx context.Context, workloads map[string]NamespaceWorkloads) error {
+	optionalComponents := map[string]NamespaceWorkloads{
+		"openshift-multus": {
+			DaemonSets: []string{"multus", "dhcp-daemon"},
+		},
+		"openshift-operator-lifecycle-manager": {
+			Deployments: []string{"olm-operator", "catalog-operator"},
+		},
+		"openshift-gateway-api": {
+			Deployments: []string{"servicemesh-operator3", "istiod-openshift-gateway-api"},
+		},
+		"kube-flannel": {
+			DaemonSets: []string{"kube-flannel-ds"},
+		},
+		"kube-proxy": {
+			DaemonSets: []string{"kube-proxy"},
+		},
+	}
```

Comment:
I think it would be a mishap for us to hardcode the optional package names and resources in the MicroShift code. The existing implementation provides a clear separation between the package and the main code by naming the resources in the scripts that are delivered together with the optional component.

Reply:
Maybe, but how would you express "ovnk should not be checked, we use flannel"?
These optional packages reside in microshift's repository anyway, so by hardcoding them we get better integration without performing hacks.